### PR TITLE
Add interactive init command (#224)

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -120,27 +129,65 @@ data Message
...
+      config <-
+        InitTemplateConfig
+          <$> prompt' "repo" repo repo
+          <*> prompt' "revision" revision revision
+          <*> prompt' "host" host host
+          <*> prompt' "port" (T.pack . show . port) port
+          <*> return (buildScript defaults)
+          <*> fmap (yesNo (restartCommand defaults) Nothing) (prompt' "Include restart command" (const "Y/n") (const "y"))
+
+      Yaml.encodeFile "hap.yml" config
```
Comment (CristhianMotoche) [inline app/Main.hs:181]: @blackheaven Can you move these lines inside the `System.Hapistrano` and associate it with a function like we do for `deploy`, `rollbar`, `deleteMaintenanceFile`, etc.? I think it's important to keep consistency. Additionally, we could add some tests for that new function.

**Review by CristhianMotoche — COMMENTED** [review]
Hi @blackheaven Thank you so much for your contribution. I left a comment there about an improvement that we can implement.

File: app/Main.hs (hunk 1 above)
Comment (blackheaven) [reply app/Main.hs]: I moved it.

However, I'm not sure which tests I can write, as it is tied to IO (`readLine`, `putStrLn`, accessing to the file system).

Comment (blackheaven) [reply app/Main.hs]: @CristhianMotoche do you have any tests you'd like to have?

Comment (CristhianMotoche) [reply app/Main.hs]: Hello @blackheaven Sorry for the delay. I think we could add a single test making sure the execution of that new command create the `hap.yml` file. Just making sure the [file does exist](https://hackage.haskell.org/package/directory-1.3.8.1/docs/System-Directory.html#v:doesFileExist).

Comment (blackheaven) [reply app/Main.hs]: Is it enough, or should I add more?

Comment (CristhianMotoche) [reply app/Main.hs]: Hi @blackheaven I think that's good enough! Thank you!

**Review by CristhianMotoche — APPROVED** [review]
LGTM!

**Comment by CristhianMotoche** [comment]
Hi @blackheaven I'm going to merge your PR. FWIW, we (as Stack Builders) started a competition related to the Hacktoberfest 2023. You can find [more information about it in our website](https://www.stackbuilders.com/hacktoberfest/). Once again. Thanks for your contribution!

**Comment by DavidMazarro** [comment]
@blackheaven Thanks for the contribution! Remember to add the following line in the PR's description so that the contribution can be considered valid for Stack Builders' Hacktoberfest 2023 event:

> As part of [Stack Builders Inc.](https://www.stackbuilders.com/) Hacktoberfest 2023 event.

**Comment by blackheaven** [comment]
Thanks

**Comment by JoseLion** [comment]
@all-contributors please add @blackheaven for code

**Pull request merged**